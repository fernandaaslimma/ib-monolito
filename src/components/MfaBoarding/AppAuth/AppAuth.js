import React, { Fragment, useState } from "react";
import { Button } from "react-bocombbm-components";
import ClickWrapper from "../../../utils/clickWrapper";
import {
  Title,
  MfaImgTutorial,
  AddAccountTutorial,
  MfaImgTutorialText,
  AppTutorial,
  MfaTextNumber
} from "./styles";
import AddAccountImg from "../../../assets/imgs/mfa_tutorial_add_account.png";
import QrCodeImg from "../../../assets/imgs/mfa_tutorial_qr_code.png";
import QrCodeSecretImg from "../../../assets/imgs/mfa_tutorial_qr_code_secret.png";
import { translate } from "../../../utils/i18n";

function AppAuth({ changeCurrentScreen }) {
  const [isLoading, setIsLoading] = useState(false);

  return (
    <Fragment>
      <Title>{translate("APP_AUTH_TUTORIAL_TITLE")}</Title>
      <AppTutorial>
        <AddAccountTutorial>
          <MfaImgTutorial src={AddAccountImg} data-test="MfaImgTutorial_1" />
          <MfaImgTutorialText>
            <MfaTextNumber>
              {translate("APP_AUTH_TUTORIAL_1_NUM")}
            </MfaTextNumber>
            {translate("APP_AUTH_TUTORIAL_1_TEXT")}
          </MfaImgTutorialText>
        </AddAccountTutorial>

        <AddAccountTutorial>
          <MfaImgTutorial src={QrCodeImg} data-test="MfaImgTutorial_2" />
          <MfaImgTutorialText>
            <MfaTextNumber>
              {translate("APP_AUTH_TUTORIAL_2_NUM")}
            </MfaTextNumber>
            {translate("APP_AUTH_TUTORIAL_2_TEXT")}
          </MfaImgTutorialText>
        </AddAccountTutorial>

        <AddAccountTutorial>
          <MfaImgTutorial src={QrCodeSecretImg} data-test="MfaImgTutorial_3" />
          <MfaImgTutorialText>
            <MfaTextNumber>
              {translate("APP_AUTH_TUTORIAL_3_NUM")}
            </MfaTextNumber>
            {translate("APP_AUTH_TUTORIAL_3_TEXT")}
          </MfaImgTutorialText>
        </AddAccountTutorial>
      </AppTutorial>
      <ClickWrapper>
        <Button
          dataTest="ShowQrCodeButton"
          onClick={() => changeCurrentScreen(4) && setIsLoading(true)}
          spacing={{ top: "xl", bottom: "l" }}
          loading={isLoading}
        >
          {translate("SHOW_QR_CODE")}
        </Button>
      </ClickWrapper>
    </Fragment>
  );
}
export default AppAuth;
