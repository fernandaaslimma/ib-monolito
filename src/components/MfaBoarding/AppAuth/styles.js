import styled from "styled-components";
import { remFontSize, rem } from "../../../styles/tools";
import { blue70, lightBlack, white } from "../../../styles/settings";

export const Title = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(24)};
  font-weight: bold;
  letter-spacing: 0.78px;
  text-align: center;
  color: ${blue70};

  margin-top: ${rem(33)};
  margin-bottom: ${rem(56)};
`;

export const AppTutorial = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
`;

export const AddAccountTutorial = styled.div`
  display: flex;
  flex-direction: column;
  margin-right: ${rem(64)};
  :last-child {
    margin-right: 0;
  }
`;

export const MfaImgTutorial = styled.img`
  width: ${rem(206)};
  height: ${rem(155)};
  object-fit: contain;
  margin-bottom: ${rem(24)};
`;

export const MfaImgTutorialText = styled.span`
  width: ${rem(211)};
  font-family: Lato;
  font-size: ${remFontSize(13)};
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: 0.35px;
  color: ${lightBlack};
  text-align: left;
`;

export const MfaTextNumber = styled(MfaImgTutorialText)`
  font-weight: 900;
  color: ${blue70};
`;

export const MfaButtonText = styled.span`
  font-family: Lato;
  font-size: ${remFontSize(13)};
  font-weight: bold;
  letter-spacing: 0.2px;
  text-align: center;
  color: ${white};
`;

export const ShowQrCodeButtonWrapper = styled.div`
  display: flex;
  margin-bottom: ${rem(14)};
  width: ${rem(179)};
`;
