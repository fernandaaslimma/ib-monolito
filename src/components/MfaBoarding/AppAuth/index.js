import AppAuth from "./AppAuth";

export default AppAuth;
