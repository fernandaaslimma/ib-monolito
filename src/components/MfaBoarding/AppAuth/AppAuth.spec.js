import React from "react";
import { shallow } from "enzyme";
import { Button } from "react-bocombbm-components";
import ClickWrapper from "../../../utils/clickWrapper";
import AppAuth from "./AppAuth";

const props = {
  current: 1,
  changeCurrentScreen: jest.fn()
};

describe("AppAuth component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<AppAuth {...props} />)).toMatchSnapshot();
  });

  it("should call changeCurrentScreen when clicked", () => {
    const shallowComponent = shallow(<AppAuth {...props} />);
    shallowComponent
      .find("Fragment")
      .find(ClickWrapper)
      .find(Button)
      .simulate("click");
    expect(props.changeCurrentScreen).toHaveBeenCalledWith(4);
  });
});
