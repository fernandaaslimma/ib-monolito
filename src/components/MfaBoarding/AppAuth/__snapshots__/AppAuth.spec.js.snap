// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAuth component should match snapshot with props 1`] = `
<Fragment>
  <styled.div />
  <styled.div>
    <styled.div>
      <styled.img
        data-test="MfaImgTutorial_1"
        src="test-file-stub"
      />
      <styled.span>
        <Styled(styled.span) />
      </styled.span>
    </styled.div>
    <styled.div>
      <styled.img
        data-test="MfaImgTutorial_2"
        src="test-file-stub"
      />
      <styled.span>
        <Styled(styled.span) />
      </styled.span>
    </styled.div>
    <styled.div>
      <styled.img
        data-test="MfaImgTutorial_3"
        src="test-file-stub"
      />
      <styled.span>
        <Styled(styled.span) />
      </styled.span>
    </styled.div>
  </styled.div>
  <ConnectWrapper>
    <Button
      dataTest="ShowQrCodeButton"
      disabled={false}
      loading={false}
      onClick={[Function]}
      spacing={
        Object {
          "bottom": "l",
          "top": "xl",
        }
      }
    />
  </ConnectWrapper>
</Fragment>
`;
